### PR TITLE
Adds GitHub workflow for building/publishing DLLs

### PR DESCRIPTION
**Pull request by ZeWaka**
Example run: https://github.com/ZeWaka/byond-tracy/runs/7315586616?check_suite_focus=true

Runs on every PR to master, and every push of mainline commits

These are maintained for 90 days by GitHub - if we want longer retention we'll need to switch to publishing Releases or refresh manually

### CODE REVIEW

**Comment by mafemergency** [comment]
if these are intended to be downloaded we need to include the licenses for lz4 and minhook in the distribution

**Comment by ZeWaka** [comment]
> if these are intended to be downloaded we need to include the licenses for lz4 and minhook in the distribution

The MinHook license will be easy, do you mind if I copy the license text in lz4.c to a text file for easier distribution?

**Comment by mafemergency** [comment]
> do you mind if I copy the license text in lz4.c to a text file for easier distribution?

i don't mind and you can find an easy copy at https://github.com/lz4/lz4/blob/dev/lib/LICENSE

**Comment by ZeWaka** [comment]
Output now in a zip with the files where they would be in the folder hierarchy. 
Ex: Output.zip -> build -> Release -> prof.dll
image

**Comment by mafemergency** [comment]
i've added licenses to the root directory you'll probably have to account for.
what i'd really like to see is a tree like this:
```
├── LICENSE-lz4.txt
├── LICENSE-minhook.txt
├── msvc
│   └── prof.dll
├── llvm
│   └── prof.dll
└── gcc
    └── prof.dll
```

if you use the ninja scripts instead of cmake you'll get the binaries in `bin` dir in this structure without the licenses
a user can then pick their poison with msvc being a good default

**Comment by ZeWaka** [comment]
ok all done

File: .github/workflows/build_publish.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+
+      - name: Build w/ LLVM
+        run: ninja -f build\llvm.ninja
+
+      # GCC
+      - name: Set up MinGW
+        uses: egor-tensin/setup-mingw@v2
+        with:
+          platform: i686
```
Comment (ZeWaka) [inline .github/workflows/build_publish.yml:49]: Mingw-w64 exists in the default windows agent, but the 64 version annoyingly won't compile the 32-bit binary no matter what i tried, so we reinstall it proper

File: .github/workflows/build_publish.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          path: |
+            C:/Program Files/LLVM
+            ./llvm
+          key: llvm-14
+
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "14.0.6"
```
Comment (ZeWaka) [inline .github/workflows/build_publish.yml:39]: latest

File: build/gcc.ninja [hunk 3]
```diff
@@ -14,7 +14,8 @@ cflags = $
     -Werror $
     -Isrc $
     -flto $
-    -D_WIN32_WINNT=0x600
+    -D_WIN32_WINNT=0x0601 $
+    -msse2
```
Comment (ZeWaka) [inline build/gcc.ninja:18]: sse2 is needed for `_mm_lfence()` and friends

File: build/gcc.ninja [hunk 4]
```diff
@@ -14,7 +14,8 @@ cflags = $
     -Werror $
     -Isrc $
     -flto $
-    -D_WIN32_WINNT=0x600
+    -D_WIN32_WINNT=0x0601 $
```
Comment (ZeWaka) [inline build/gcc.ninja:17]: byond doesn't support xp anymore anyways, 7 is more sane

File: src/microtracy/server.c [hunk 5]
```diff
@@ -52,7 +52,7 @@ extern int utracy_server_init(void) {
 		addrinfo->ai_protocol,
 		NULL,
 		0,
-		WSA_FLAG_NO_HANDLE_INHERIT
+		0
```
Comment (ZeWaka) [inline src/microtracy/server.c:55]: image

**Pull request merged**